src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
+        }
+        Session session = this.factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error in adding actor", e);
+        } finally {
+            session.close();
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        try (Session session = this.factory.openSession()) {
+            return Optional.ofNullable(session.get(Actor.class, id));
```

Comment:
Using try-with-resources for the session in the `get` method is good practice to ensure the session is closed after the operation is complete. Keep it up.